packages/desktop-client/src/components/reports/spreadsheets/default-spreadsheet.tsx

```diff
@@ -106,388 +87,104 @@ export default function createSpreadsheet(
-          [groupByLabel]: splt.id,
-        })
-        //Apply month range filters
-        .filter({
-          $and: [
-            { date: { $transform: '$month', $gte: start } },
-            { date: { $transform: '$month', $lte: end } },
-          ],
-        })
-        //Show assets or debts
-        .filter(
-          name === 'assets' ? { amount: { $gt: 0 } } : { amount: { $lt: 0 } },
-        );
-
-      return query
-        .groupBy({ $month: '$date' })
-        .select([
-          { date: { $month: '$date' } },
-          { [name]: { $sum: '$amount' } },
-        ]);
-    }
-
-    const graphData = await Promise.all(
-      groupByList.map(async splt => {
-        let [starting, assets, debts] = await Promise.all([
-          runQuery(
-            q('transactions')
-              .filter(
-                !hidden && {
-                  $and: [
-                    {
-                      'account.offbudget': false,
-                      $or: [
-                        {
-                          'category.hidden': false,
-                          category: null,
-                        },
-                      ],
-                    },
-                  ],
-                  $or: [
-                    {
-                      'payee.transfer_acct.offbudget': true,
-                      'payee.transfer_acct': null,
-                    },
-                  ],
-                },
-              )
-              .filter(
-                splt.uncat_id === '2'
-                  ? {
-                      'payee.transfer_acct.closed': false,
-                    }
-                  : {
-                      'payee.transfer_acct': null,
-                      'account.offbudget': splt.offBudget
-                        ? splt.offBudget
-                        : false,
-                    },
-              )
-              .filter(
-                selectedCategories && {
-                  $or: categoryFilter.map(category => ({
-                    category: category.id,
-                  })),
-                },
-              )
-              .filter({
-                [conditionsOpKey]: [...filters],
-                [groupByLabel]: splt.id,
-              })
-              .filter({
-                $and: [{ date: { $lt: start + '-01' } }],
-              })
-              .calculate({ $sum: '$amount' }),
-          ).then(({ data }) => data),
-
-          runQuery(makeQuery(splt, 'assets')).then(({ data }) => data),
-          runQuery(makeQuery(splt, 'debts')).then(({ data }) => data),
-        ]);
-
-        return {
-          id: splt.id,
-          uncat_id: splt.uncat_id,
-          name: splt.name,
-          starting,
-          hidden: splt.hidden,
-          assets: index(assets, 'date'),
-          debts: index(debts, 'date'),
-        };
-      }),
-    );
+    let assets: QueryDataEntity[];
+    let debts: QueryDataEntity[];
+
+    [assets, debts] = await Promise.all([
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏same problem here
```suggestion
    const [assets, debts] = await Promise.all([
```